students/Gabe_Lamberth/lesson09/mailroom5.py

```diff
@@ -1,25 +1,196 @@
 #!/usr/bin/env python3
-from collections import *
 
+from pathlib import Path
+from random import randint
+import sys
 
-class Donor(dict):
 
-    def __getattr__(self, item):
-        if item in self:
-            return self[item]
-        else:
-            raise AttributeError(f'No such attribute: {item} ')
+class Donor(object):
 
-    def __setattr__(self, key, value):
-        self[key] = value
+    def __init__(self, dictionary):
+        self.__dict__ = dictionary
 
-    def __delattr__(self, item):
-        if item in self:
-            del self[item]
-        else:
-            raise AttributeError(f'No such attribute: {item}')
+    def __setitem__(self, key, item):
+        self.__dict__[key] = item
+
+    def __getitem__(self, key):
+        return self.__dict__[key]
+
+    def __repr__(self):
+        return repr(self.__dict__)
+
+    def __len__(self):
+        return len(self.__dict__)
+
+    def __delitem__(self, key):
+        del self.__dict__[key]
+
+    def clear(self):
+        return self.__dict__.clear()
+
+    def copy(self):
+        return self.__dict__.copy()
+
+    def has_key(self, k):
+        return k in self.__dict__
+
+    def update(self, *args, **kwargs):
+        return self.__dict__.update(*args, **kwargs)
+
+    def keys(self):
+        return ', '.join(self.__dict__.keys())
+
+    def values(self):
+        return self.__dict__.values()
+
+    def items(self):
+        return self.__dict__.items()
+
+    def pop(self, *args):
+        return self.__dict__.pop(*args)
+
+    def __contains__(self, item):
+        return item in self.__dict__
+
+    def __iter__(self):
+        return iter(self.__dict__)
 
     def append(self, key, value):
-        self[key].append(value)
+        self.__setitem__(key, value)
```

Comment:
You likely didn't need to override all this functions for the Donor class, as the `patron` dictionary that you are passing in is already a `dict` it would have all this functionality. `.keys()` and `.items()` appear to be the only functions you actually use from this list of functions you created.  I'm not sure if this would work, but i might make `class Donar(dict):` to have the `Donor` class inherit from the a `dict` object.